packages/world/src/modules/core/CoreModule.sol

```diff
@@ -97,34 +109,62 @@ contract CoreModule is Module {
   }
 
   /**
-   * @notice Register the CoreSystem in the World.
-   * @dev Uses the CoreSystem's `registerSystem` implementation to register itself on the World.
+   * @notice Register the core systems in the World.
+   */
+  function _registerCoreSystems() internal {
+    _registerSystem(accessManagementSystem, ACCESS_MANAGEMENT_SYSTEM_ID);
+    _registerSystem(balanceTransferSystem, BALANCE_TRANSFER_SYSTEM_ID);
+    _registerSystem(batchCallSystem, BATCH_CALL_SYSTEM_ID);
+    _registerSystem(coreRegistrationSystem, CORE_REGISTRATION_SYSTEM_ID);
+  }
+
+  /**
+   * @notice Register the core system in the World.
+   * @dev Uses the CoreRegistrationSystem's `registerSystem` implementation to register the system on the World.
    */
-  function _registerCoreSystem() internal {
+  function _registerSystem(address target, ResourceId systemId) internal {
     WorldContextProviderLib.delegatecallWithContextOrRevert({
       msgSender: _msgSender(),
       msgValue: 0,
-      target: coreSystem,
-      callData: abi.encodeCall(WorldRegistrationSystem.registerSystem, (CORE_SYSTEM_ID, CoreSystem(coreSystem), true))
+      target: coreRegistrationSystem,
+      callData: abi.encodeCall(WorldRegistrationSystem.registerSystem, (systemId, WorldContextConsumer(target), true))
     });
   }
 
   /**
-   * @notice Register function selectors for all CoreSystem functions in the World.
+   * @notice Register function selectors for all core system functions in the World.
    * @dev Iterates through known function signatures and registers them.
    */
   function _registerFunctionSelectors() internal {
-    string[19] memory functionSignatures = [
+    string[3] memory functionSignaturesAccessManagement = [
       // --- AccessManagementSystem ---
       "grantAccess(bytes32,address)",
       "revokeAccess(bytes32,address)",
-      "transferOwnership(bytes32,address)",
+      "transferOwnership(bytes32,address)"
```

Comment:
just thinking aloud but I wonder if it'd be worth our codegen creating a solidity file for system function selectors, so we can import from that instead of maintaining a list here